### PR TITLE
jsonrpc

### PR DESCRIPTION
**Pull request by debris**
- implemented few more jsonrpc methods
- jsonrpc serialization tests
- added transaction querying api to blockchain.rs

### CODE REVIEW

**Comment by gavofyork** [comment]
this changes the database format (inserts additional stuff into extras db). did you alter the DB format to force re-sync?

**Comment by debris** [comment]
how to force re-sync?

**Comment by gavofyork** [comment]
@arkpar added the database versioning a few commits ago...

**Comment by debris** [comment]
re-sync forced

File: util/src/journaldb.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ impl Clone for JournalDB {
 const LAST_ERA_KEY : [u8; 4] = [ b'l', b'a', b's', b't' ]; 
 const VERSION_KEY : [u8; 4] = [ b'j', b'v', b'e', b'r' ]; 
 
-const DB_VERSION: u32 = 1;
+const DB_VERSION: u32 = 2;
```
Comment (arkpar) [inline util/src/journaldb.rs]: That's the state db version. You need to change client version in client.rs instead

**Pull request merged**